api/v1alpha1/mariadb_types.go

```diff
@@ -478,6 +478,11 @@ type TLS struct {
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
 	Required *bool `json:"required,omitempty"`
+	// Mutual specifies whether TLS must be mutual between server and client.
+	// It enabled by default.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	Mutual *bool `json:"mutual,omitempty"`
```

Comment:
I think this would create some confussion to the `MariaDB` users, if they set it to `false` the connections would present the client certificate anyway. 

Could you create a new `ExternalTLS` struct, which would embed  the current `TLS` struct and additionally inclue the `Mutual` field? This external struct, as you probably guessed, will be used exclusively in the `ExternalMariaDB`.